packages/core/src/node/backend-application-gzip-middleware.ts

```diff
@@ -0,0 +1,39 @@
+/********************************************************************************
+ * Copyright (C) 2017 TypeFox and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+import * as express from 'express';
+import * as fs from 'fs-extra';
+import * as mimeType from 'mime-types';
+
+export const serveGzip = function (req: express.Request, res: express.Response, next: express.NextFunction) {
```

Comment:
Everything should be provided inversify to be easily customizable, i.e. it should be a protected method of an injectable class.

Reply:
Hmm, I don't quite catch you (sorry for being really dumb, its really my first time in the codebase), can you cite an example?

Reply:
We use inversify that extenders can rebind and override implementations. `serverGzip` not in inversify context and it is not possible to customize it.

Could you turn it into a protected method on `BackendApplication`?

Reply:
Ah I see, I'll do it right away